src/main/java/org/kiwiproject/dropwizard/error/dao/jdk/JdbcApplicationErrorDao.java

```diff
@@ -0,0 +1,322 @@
+
+        var sql = "insert into application_errors" +
+                " (description, exception_type, exception_message, exception_cause_type, exception_cause_message," +
+                " stack_trace, host_name, ip_address, port)" +
+                " values (?, ?, ?, ?, ?, ?, ?, ?, ?)";
+
+        try (var conn = connection(); var ps = conn.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            ps.setString(1, newError.getDescription());
+            ps.setString(2, newError.getExceptionType());
+            ps.setString(3, newError.getExceptionMessage());
+            ps.setString(4, newError.getExceptionCauseType());
+            ps.setString(5, newError.getExceptionCauseMessage());
+            ps.setString(6, newError.getStackTrace());
+            ps.setString(7, newError.getHostName());
+            ps.setString(8, newError.getIpAddress());
+            ps.setInt(9, newError.getPort());
+
+            var count = ps.executeUpdate();
+            checkState(count == 1, "Insert count should be one, but is: %s", count);
+
+            try (ResultSet generatedKeys = ps.getGeneratedKeys()) {
+                nextOrThrow(generatedKeys);
+                return generatedKeys.getLong(1);
+            }
+        } catch (SQLException e) {
+            throw new ApplicationErrorJdbcException(e);
+        }
+    }
+
+    @Override
+    public void incrementCount(long id) {
+        var sql = "update application_errors" +
+                " set num_times_occurred = num_times_occurred + 1, updated_at = current_timestamp" +
+                " where id = ?";
+
+        try (var conn = connection(); var ps = conn.prepareStatement(sql)) {
+            ps.setLong(1, id);
+
+            var count = ps.executeUpdate();
+            checkState(count == 1, "Unable to increment count. No ApplicationError found with id %s", id);
+        } catch (SQLException e) {
+            throw new ApplicationErrorJdbcException(e);
+        }
+    }
+
+    @Override
+    public long insertOrIncrementCount(ApplicationError error) {
+        checkNotNull(error.getDescription(), "Error description cannot be null");
+
+        var errors = getUnresolvedErrorsByDescriptionAndHost(error.getDescription(), error.getHostName());
+
+        if (errors.isEmpty()) {
+            return insertError(error);
+        }
+
+        var existingError = first(errors);
+        incrementCount(existingError.getId());
+        return existingError.getId();
+    }
+
+    @Override
+    public ApplicationError resolve(long id) {
+        var sql = "update application_errors set resolved = true, updated_at = current_timestamp where id = ?";
+
+        try (var conn = connection(); var ps = conn.prepareStatement(sql)) {
+            ps.setLong(1, id);
+
+            var count = ps.executeUpdate();
+            checkState(count == 1, "Unable to resolve. No ApplicationError found with id %s", id);
```

Comment:
Is it worth extracting a method that executes the update and verifies an expected number of rows, along with an error message if the check fails?

`updateAndCheckCount(ps, 1, "did not update one row for id {}", id);`

Reply:
Added this as a discussion in kiwi: https://github.com/kiwiproject/kiwi/discussions/1075